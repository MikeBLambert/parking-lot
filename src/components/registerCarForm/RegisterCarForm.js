import React, { PureComponent, Fragment } from 'react';

export default class RegisterCarForm extends PureComponent {

  state = {
    make: 'Toyota',
    model: '',
    plate: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {


    return (
      <Fragment>
        <h1>Register a New Car</h1>
        <form>
          <label htmlFor='make'>Make</label>
          <input id='make' type='text' name='make' onChange={this.handleChange}/>

          <label htmlFor='model'>Model</label>
          <input id='model' type='text' name='model' onChange={this.handleChange}/>

          <label htmlFor='plate'>Plate Info</label>
          <input id='plate' type='text' name='plate' onChange={this.handleChange}/>

          <button onClick={this.handleSubmit}>Register</button>
        </form>
      </Fragment>
    );
  }
}
