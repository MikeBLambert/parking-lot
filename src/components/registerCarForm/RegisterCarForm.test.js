import React from 'react';
import { shallow } from 'enzyme';
import RegisterCarForm from './RegisterCarForm';

describe('RegisterCarForm component', () => {
  it('renders a component', () => {
    const wrapper = shallow(<RegisterCarForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has 3 input forms', () => {
    const wrapper = shallow(<RegisterCarForm />);
    expect(wrapper.find('input')).toHaveLength(3);
  });
});
