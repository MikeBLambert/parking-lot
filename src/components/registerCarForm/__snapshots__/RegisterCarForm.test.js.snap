// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterCarForm component renders a component 1`] = `<div />`;
